components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -336,10 +335,19 @@ int PickRoulette(const ArticleInfos& articles) {
 mojom::FeedItemMetadataPtr PickAndRemove(ArticleInfos& articles,
                                          PickArticles picker) {
   auto index = picker.Run(articles);
+
+  // If there were no valid articles to pick the index will be |-1|.
   if (index == -1) {
     return nullptr;
   }
 
+  if (index < 0 || static_cast<uint32_t>(index) >= articles.size()) {
```

Comment:
`base::strict_cast<ArticleInfos::size_type>`

Reply:
`std::vector<T>::size_type` is typically `size_t` and I think a lot of chromium code makes that assumption so that is probably ok here too